projects/plugins/jetpack/modules/comments/base.php

```diff
@@ -41,38 +52,39 @@ protected function setup_filters() {
 	 * Is this a Highlander POST request?
 	 * Optionally restrict to one or more credentials slug (facebook, twitter, ...)
 	 *
-	 * @param string Comment credentials slug
-	 * @param ...
+	 * @param mixed ...$args Comments credentials slugs.
 	 * @return false|string false if it's not a Highlander POST request.  The matching credentials slug if it is.
 	 */
-	function is_highlander_comment_post( ...$args ) {
-		if ( empty( $_POST['hc_post_as'] ) ) {
+	public function is_highlander_comment_post( ...$args ) {
+		if ( empty( $_POST['hc_post_as'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing -- we are not using the raw $_POST data
```

Comment:
Were you going to update all these comments to something like "Nonce verification should happen in Jetpack_Comments::pre_comment_on_post()" instead?

That's here, line 65, line 71, line 250, line 252, line 263, and maybe line 266.

Reply:
On line 525 I'd added this: `@todo We do need to add a nonce check here - internal ref for details: p1645643468937519/1645189749.180299-slack-C02HQGKMFJ8` (in function `watch_comment_parent`) thinking one mention would cover all other examples, but now you mention it I can see it does make sense to mention it for every occurrence. Also I realize the link itself makes it difficult for anyone without that access to know what to do, so I can also add a more detailed explanation (well, like what you've suggested). Is it still worth including the link, maybe even just in one example?

Reply:
More details wouldn't hurt, but in practice it's going to have to be an a11n working on it since it needs changes to wpcom so that part probably doesn't matter.

The comment on line 535 in the other file is fine, what I was referring to here are the "we are not using the raw $_POST data" statements on the lines I mentioned in this file.

Reply:
Ah of course, you linked to base.php, and I'd been looking at comments.php. Yes I'll definitely go in and change those. As I look at this more, I'm thinking of making use of `// phpcs:disable` and `// phpcs:enable` for the specific sniff, from around lines 250 - 266 since there are a few in one small area.

Reply:
It turns out the first few nonce messages were also in just one function, so I ended up adding two separate `// phpcs:disable` and `// phpcs:enable` blocks, hoping that will be cleaner but also sufficient.